File: content/snippets/cpp/exercises/beginner/factorial.cpp
#include <iostream>
using namespace std;

long long factorial(int n) {
    long long r = 1;
    for (int i = 2; i <= n; ++i) r *= i;
    return r;
}

int main() {
    int n;
    if (!(cin >> n)) n = 5;
    cout << factorial(n) << "\n";
    return 0;
}

File: content/snippets/python/exercises/beginner/factorial.py
def factorial(n: int) -> int:
    r = 1
    for i in range(2, n+1):
        r *= i
    return r

if __name__ == '__main__':
    try:
        n = int(input().strip())
    except Exception:
        n = 5
    print(factorial(n))
